include/mppic/parameters_handler.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 FastSense, Samsung Research
+
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <type_traits>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_util/node_utils.hpp"
+
+namespace mppi
+{
+
+enum class ParameterType
+{
+  Dynamic,
+  Static
+};
+
+class ParametersHandler
+{
+public:
+  using get_param_func_t = void (const rclcpp::Parameter & param);
+  using post_callback_t = void ();
+  using pre_callback_t = void ();
+
+  ParametersHandler() = default;
+  ParametersHandler(const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent)
+  {
+    node_ = parent;
+    auto node = node_.lock();
+    logger_ = node->get_logger();
+  }
+
+  void start()
+  {
+    auto node = node_.lock();
+    on_set_param_handler_ = node->add_on_set_parameters_callback(
+      std::bind(
+        &ParametersHandler::dynamicParametersCallback, this, std::placeholders::_1));
+  }
+
+  rcl_interfaces::msg::SetParametersResult dynamicParametersCallback(
+    std::vector<rclcpp::Parameter> parameters)
+  {
+    rcl_interfaces::msg::SetParametersResult result;
+
+    std::lock_guard<std::mutex> lock(parameters_change_mutex_);
+
+    for (auto & pre_cb : pre_callbacks_) {
+      pre_cb();
+    }
+
+    for (auto & param : parameters) {
+      const std::string & param_name = param.get_name();
+
+      if (auto callback = get_param_callbacks_.find(param_name);
+        callback != get_param_callbacks_.end())
+      {
+        RCLCPP_INFO(logger_, "Parameter %s found", param_name.c_str());
```

Comment:
Probably should be debug logs and add more context like what the value of it is being dynamically set

Reply:
I'm not sure this must be debug log, context in callback function

Reply:
OK - more info then would be good then to give the parameter name and value for setting so its all the context needed.

Reply:
Log about what param and what value are in callback function

Reply:
https://github.com/artofnothingness/mppic/blob/f/params/include/mppic/parameters_handler.hpp#L149

Mb remove param found part